scene/surface/geometry/Isosurface.cpp

```diff
@@ -26,18 +32,30 @@ void Isosurface::commit()
   auto og = osprayGeometry();
 
   if (m_isovalue) {
-    auto iv = m_isovalue->osprayData();
-    ospSetParam(og, "isovalue", OSP_DATA, &iv);
+    assert(m_isovalue->elementType() == ANARI_FLOAT32);
```

Comment:
Nice, good to check. However, `assert` is quite intrusive and sometime does nothing (when `NDEBUG`), all other error checking is using the `reportMessage` mechanism.

Reply:
I amended to the commit and just added those checks for `elementType()` and `size()` to the `if` statement. `size` is what my viewer trips over, under certain conditions it sets an empty `isovalue` array on the geometry and in this case it should be considered invalid. The `elementType` check was primarily due to unpacking, but I guess the check doesn't hurt.